examples/js/loaders/MMDLoader.js

```diff
@@ -1017,14 +1019,19 @@
 					if ( material.textureIndex !== - 1 ) {
 
 						params.map = this._loadTexture( data.textures[ material.textureIndex ], textures );
+						params.map.name = material.name; // add fileName for easier debugging
+
+						params.map.fileName = data.textures[ material.textureIndex ];
```

Comment:
Please remove debugging code.

Reply:
This actually turns out to be a needed thing for me to enable `.normalMap`.

According to the spec I found in https://gist.github.com/felixjones/f8a06bd48f9da9a4539f#texture
> Some implementations will change texture behaviour based on file name, such as using "_s" as sphere map or "_n" as normal map or may have texture entries that do not even reference files but reference some other data.

There seems no standard of mapping between color map and normal map of PMX files or normal map index in Material Data (in my reference model its `_N`, and it not even be listed in Textures fields in PMX file). So I need to append it by my self after model loaded like:

```js
const normalMapParing = { 
  'tex\\cloth.dds': 'tex\\cloth_N.png',
  'tex\\cloth2.dds': 'tex\\cloth23_N.png',
  'tex\\cloth3.dds': 'tex\\cloth23_N.png', // cloth2 and cloth3 share same normal file
  'tex\\cloth4.dds': null,
};

mmdLoader.load(mesh => {
  mesh.material.forEach(m => { 
    const normalMapFilename = normalMapParing[m.map.fileName];
    if (normalMapFilename) { 
      m.normalMap = textureLoader.load(normalMapFilename);
      m.normalMap.flipY = false;
    } 
    if (m.uniforms) { 
      m.uniforms.normalMap.value = m.normalMap;
    } 
  });
})
```

Or maybe I could move it to material's `.userData`? Something like this:
```js
material.userData.MMD = {
  mapFileName: data.textures[ material.textureIndex ],
};
```

Reply:
@takahirox sorry if this bothers you. Just to notify you in case you miss this comment.

Reply:
Thanks, I didn't notice the comment.

Interesting... I didn't know that specification.

> Or maybe I could move it to material's `.userData`?

Yes, using `.userData` should be much better. And please add comment why we save the file names.

Another option may be we don't do nothing because file name can be gotten  with `material.map.image.src`. But a disadvantage of it is if user sets `ImageBitmapLoader` as handers to manager, no file name in `.src`.

> not even be listed in Textures fields in PMX file

Curious to know, how loaders or users can know the existence of such textures?

Reply:
> Another option may be we don't do nothing because file name can be gotten with `material.map.image.src`. But a disadvantage of it is if user sets `ImageBitmapLoader` as handers to manager, no file name in `.src`.

Also in my case the texture file is `.dds` and loaded as `CompressedTexture`, another condition the wouldn't have file name.

> Curious to know, how loaders or users can know the existence of such textures?

For my situation, I need to inspect the model folder to know there is normal map files in it. :(

Reply:
Also the paring between color map files and normal map files are quite arbitrary.
So for now I think we just leave it for MMDLoader users to process it afterwards instead of handle it in MMDLoader.

Reply:
> For my situation, I need to inspect the model folder to know there is normal map files in it. :(

Oh

> So for now I think we just leave it for MMDLoader users to process it afterwards instead of handle it in MMDLoader.

Then, yes. I think it's a right direction.